src/cbor/CBOREncoder.cpp

```diff
@@ -45,24 +46,68 @@ CborError CBOREncoder::encode(PropertyContainer & property_container, uint8_t *
    */
   CborError error = CborNoError;
   int num_encoded_properties = 0;
-  std::for_each(property_container.begin(),
-                property_container.end(),
-                [lightPayload, &arrayEncoder, &error, &num_encoded_properties](Property * p)
-                {
-                  if (p->shouldBeUpdated() && p->isReadableByCloud())
-                  {
-                    error = p->append(&arrayEncoder, lightPayload);
-                    if(error == CborNoError)
-                      num_encoded_properties++;
-                    else
-                      return;
-                  }
-                });
-  if ((CborNoError != error) && 
-      (CborErrorOutOfMemory != error))
+  int num_checked_properties = 0;
+  static int encoded_properties_message_limit = CBOR_ENCODER_NO_PROPERTIES_LIMIT;
+
+  if(current_property_index >= property_container.size())
+    current_property_index = 0;
+
+  PropertyContainer::iterator iter = property_container.begin();
+  std::advance(iter, current_property_index);
+
+  for(; iter != property_container.end(); iter++)
+  {
+    Property * p = * iter;
```

Comment:
I'm trying to make up my mind whether or not this "duplication" of `iter` is really necessary. Possibly if you rename `iter` to `prop_iter` and then access via `(*prop_iter)->`. Granted, this syntax (`(*exp)->`) is somewhat ugly. I leave it for you to decide :shrug:

Reply:
I personally i don't like either of them, but i like a little bit more `Property * p = * iter;` despite the variable duplication.

Reply:
Works for me 👍 😉